### PR TITLE
feat: add onStepError configuration option #143

### PR DESCRIPTION
**Pull request by amiceli**

### CODE REVIEW

File: src/vitest/__tests__/on-step-error.spec.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        onTestFailed: vi.fn().mockImplementation((callback: () => void) => {
+            onTestFailedCallback = callback
+        }),
+    }
+})
+
+describe('Use onStepError', () => {
+    describe('Scenario', () => {
+        const feature = FeatureContentReader.fromString([
+            `Feature: onStepError options`,
```
Comment (Odonno) [inline src/vitest/__tests__/on-step-error.spec.ts:27]: note: you can write multiline string with template literals, avoiding the need for an array.

**Review by Odonno — APPROVED** [review]
Looks pretty good to me.

File: src/vitest/__tests__/on-step-error.spec.ts (hunk 1 above)
Comment (amiceli) [reply src/vitest/__tests__/on-step-error.spec.ts:27]: Yes I know, but `FeatureContentReader` uses `GherkinParser.addLine`.
I prefer pass an array instead a string to split before `for of`.

File: src/vitest/describe/describeBackground.ts [hunk 2]
```diff
@@ -20,6 +21,7 @@ export function createBackgroundDescribeHandler({
     backgroundCallback,
 }: DescribeScenarioArgs): () => void {
     const backgroundStepsToRun: ScenarioSteps[] = []
+    const cucumberConfiguration = getVitestCucumberConfiguration()
```
[the remark this answers is not in the document]
Comment (amiceli) [reply src/vitest/describe/describeBackground.ts:24]: Yes, I will rename it.

**Comment by amiceli** [comment]
Ok, I will merge, publish and update doc.

**Pull request merged**